src/core/execution/FakeHumanExecution.ts

```diff
@@ -552,6 +553,52 @@ export class FakeHumanExecution implements Execution {
           }
 
           // TODO: Cities and factories should consider train range limits
+          return w;
+        };
+      case UnitType.SAMLauncher:
+        const structureTiles: TileRef[] = [];
+        for (const unit of this.player.units()) {
+          switch(unit.type()) {
+            case UnitType.City:
+            case UnitType.Factory:
+            case UnitType.MissileSilo:
+            case UnitType.Port:
+            case UnitType.DefensePost:
+            case UnitType.SAMLauncher:
+              structureTiles.push(unit.tile());
+
+          }
+        }
+        return (tile) => {
+          if (this.player === null) throw new Error("Not initialised.");
+          let w = 0;
+
+          // According to the inner radius of a hydrogen bomb around each structure:
+          // - Increase points for how much closer the tile is to the structure
+          // in relation to the perimeter.
+          // - Deecrease points for how much farther the tile is to the structure
+          // in relation to the perimeter.
+          // This pushes the nation to focus SAMs in a "goldilocks zone" that is neither
+          // too close nor too far from their structure clusters.
+          const hydrogenSpacing = this.mg.config().nukeMagnitudes(UnitType.HydrogenBomb).inner;
+
+          for (const certainTile of structureTiles) {
+            const distanceVector = [mg.x(certainTile) - mg.x(tile), mg.y(certainTile) - mg.y(tile)];
+            const distanceMagnitude = (distanceVector[0] ** 2 + distanceVector[1] ** 2);
```

Comment:
The array allocation and exponentiation operations here are unnecessary.
```suggestion
            const dx = mg.x(certainTile) - mg.x(tile);
            const dy = mg.y(certainTile) - mg.y(tile);
            const distanceMagnitude = dx * dx + dy * dy;
```